{{cookiecutter.project_dirname}}/scripts/deploy/vault.sh

```diff
@@ -1,14 +1,18 @@
 #!/bin/sh -e
 
-export VAULT_TOKEN=`curl --silent --request POST --data "role=${VAULT_ROLE}" --data "jwt=${CI_JOB_JWT_V2}" ${VAULT_ADDR%/}/v1/auth/gitlab-jwt/login | jq -r .auth.client_token`
+vault_token=$(curl --silent --request POST --data "role=${VAULT_ROLE}" --data "jwt=${CI_JOB_JWT_V2}" "${VAULT_ADDR%/}"/v1/auth/gitlab-jwt/login | jq -r .auth.client_token)
+
+secrets_data="{}"
 
 for secret_path in ${VAULT_SECRETS}
 do
-    secret_var_file=${TERRAFORM_VARS_DIR}/`echo ${secret_path} | tr / -`.json
-    curl --silent --header "X-Vault-Token: ${VAULT_TOKEN}" ${VAULT_ADDR%/}/v1/${PROJECT_SLUG}/${VAULT_SECRETS_PREFIX}/${secret_path} | jq -r ".data // {}" > ${secret_var_file}
-    var_files="${var_files} -var-file=${secret_var_file}"
+    secret_data=$(curl --silent --header "X-Vault-Token: ${vault_token}" "${VAULT_ADDR%/}"/v1/"${PROJECT_SLUG}"/"${VAULT_SECRETS_PREFIX}"/"${secret_path}" | jq -r '.data // {}') || secret_data="{}"
+    secrets_data=$(echo "${secrets_data}" | jq --argjson new_data "${secret_data}" '. * $new_data')
 done
 
-if [ "${TERRAFORM_BACKEND}" == "terraform-cloud" ]; then
-    export TFC_TOKEN=`curl --silent --header "X-Vault-Token: ${VAULT_TOKEN}" ${VAULT_ADDR%/}/v1/${PROJECT_SLUG}-tfc/creds/default | jq -r .data.token `
+echo "${secrets_data}" > "${TERRAFORM_VARS_DIR%/}"/vault-secrets.tfvars.json
+
+if [ "${TERRAFORM_BACKEND}" = "terraform-cloud" ]; then
+    TFC_TOKEN=$(curl --silent --header "X-Vault-Token: ${vault_token}" "${VAULT_ADDR%/}"/v1/"${PROJECT_SLUG}"-tfc/creds/default | jq -r .data.token)
+    export TFC_TOKEN
```

Comment:
consider to add an `unset` for sensitive environment variables at point where they become no longer necessary

Reply:
GitLab runner deals with this already, there are a lot of sensitive env vars involved in a single run of a CI job (e.g. the JWT token itself), and they are all handled correctly by the runner. I would not worry about this or anything of the sort.